kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/ksp/shadedUtil.kt

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi.kotlin.codegen.ksp
+
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.symbol.KSAnnotated
+import com.google.devtools.ksp.symbol.KSAnnotation
+import com.google.devtools.ksp.symbol.KSClassDeclaration
+import com.google.devtools.ksp.symbol.KSType
+import com.google.devtools.ksp.symbol.KSValueArgument
+import java.lang.reflect.InvocationHandler
+import java.lang.reflect.Method
+import java.lang.reflect.Proxy
+import java.util.concurrent.ConcurrentHashMap
+import kotlin.reflect.KClass
+
+/*
+ * Copied experimental utilities from KSP.
+ */
+
+/**
+ * Find a class in the compilation classpath for the given name.
+ *
+ * @param name fully qualified name of the class to be loaded; using '.' as separator.
+ * @return a KSClassDeclaration, or null if not found.
+ */
+internal fun Resolver.getClassDeclarationByName(name: String): KSClassDeclaration? =
+  getClassDeclarationByName(getKSNameFromString(name))
+
+internal fun <T : Annotation> KSAnnotated.getAnnotationsByType(annotationKClass: KClass<T>): Sequence<T> {
+  return this.annotations.filter {
+    it.shortName.getShortName() == annotationKClass.simpleName && it.annotationType.resolve().declaration
```

Comment:
any idea what the utility of the short name check is?

Reply:
Simple early check before doing a more expensive resolve() call in the second check